CMakeLists.txt

```diff
@@ -139,10 +139,11 @@ if (MFEM_USE_CUDA)
       set(CMAKE_CUDA_HOST_LINK_LAUNCHER ${CMAKE_CXX_COMPILER})
    endif()
    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} ${CUDA_FLAGS}")
+   find_package(CUDAToolkit REQUIRED)
    set(CUSPARSE_FOUND TRUE)
-   set(CUSPARSE_LIBRARIES "cusparse")
+   get_target_property(CUSPARSE_LIBRARIES CUDA::cusparse LOCATION)
```

Comment:
I think it is a bit better to just set `CUSPARSE_LIBRARIES` to be the target `CUDA::cusparse` itself -- this way more information about the target will be propagated, not just the location.

Reply:
I thought the same at first, but this would require users using the exported `MFEMTargets.cmake` file to `find_package(CUDAToolkit)` before `find_package(mfem)` to resolve the targets `CUDA::cusparse` and `CUDA::cublas`, which I worry might lead to a lot of confusion. I think the "fully modern" CMake approach would be to have the `MFEMTargets.cmake` file locate all the unresolved targets but this is a pretty substantial change I'm pretty sure from the current structure and use. So, I decided to keep it as libraries for now.

Reply:
Okay, sounds good. We can revisit if we encounter issues in the future.